drools-game-engine-horserace-model/src/main/java/org/drools/game/horserace/model/Checkpoint.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.drools.game.horserace.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Samuel
+ */
+public class Checkpoint {
+
+    private String id;
+    private Checkpoint required;
+    private List<String> playersCrossedCheckpoint = new ArrayList<>();
```

Comment:
Would you prefer that it's attached to the player, or checkpoint?

declare CheckpointBundle
    checkpoint: Checkpoint @key
    passedthrough : List<String> @key
end

or


declare PlayerBundle
    player: Player @key
    checkPointsPassedThrough: List<Checkpoint> @key
end